### PR TITLE
Expose Attachment Settings in the API

### LINKED ISSUE #12368: [Enhancement] [API] expose upload-limits on settings endpoint
https://github.com/go-gitea/gitea/blob/master/custom/conf/app.example.ini#L742-L752

(Issue/Pull/Releaes - Attachemnts) ...

### PR DESCRIPTION
**Pull request by 6543**
close #12368

### CODE REVIEW

**Comment by silverwind** [comment]
Will this conflict https://github.com/go-gitea/gitea/pull/12465? There we will add a new `repository.release.ALLOWED_TYPES` setting specific to releases.

**Comment by 6543** [comment]
@silverwind looks not - but i think it is good to wait for your pull since then it would make more sence to add some things to other api endpoints ... (allowed for Repo specific ...)

**Review by zeripath — APPROVED** [review]

File: modules/structs/settings.go [hunk 1]
```diff
@@ -14,3 +14,11 @@ type GeneralRepoSettings struct {
 type GeneralUISettings struct {
 	AllowedReactions []string `json:"allowed_reactions"`
 }
+
+// GlobalSettings contains global settings witch dont have its own category exposed by API
+type GlobalSettings struct {
```
Comment (mrsdizzie) [inline modules/structs/settings.go]: Why GlobalSettings and not "Attachment" ? I think it should follow the config of /settings/category just like ui, repo

 https://docs.gitea.io/en-us/config-cheat-sheet/#attachment-attachment

**Comment by 6543** [comment]
@mrsdizzie done

**Review by mrsdizzie — APPROVED** [review]

**Pull request merged**